src/bin/dolos/common.rs

```diff
@@ -130,26 +139,86 @@ pub fn setup_tracing(config: &LoggingConfig) -> miette::Result<()> {
         filter = filter.with_target("tower_http", level);
     }
 
-    #[cfg(not(feature = "debug"))]
-    {
-        tracing_subscriber::registry()
-            .with(tracing_subscriber::fmt::layer())
-            .with(filter)
-            .init();
-    }
+    // Check if telemetry should be enabled (default to true if not specified)
+    let include_telemetry = config.include_telemetry.unwrap_or(true);
+
+    if include_telemetry {
+        // Use telemetry config if provided, otherwise use defaults
+        let default_telemetry_config = TelemetryConfig::default();
+        let telemetry_config = config.telemetry.as_ref().unwrap_or(&default_telemetry_config);
+
+        let exporter = opentelemetry_otlp::LogExporter::builder()
+            .with_http()
+            .with_protocol(Protocol::HttpJson)
+            .with_endpoint(&telemetry_config.endpoint)
+            .with_timeout(Duration::from_secs(3))
+            .build().unwrap();
+
+        let provider = sdklog::SdkLoggerProvider::builder()
+            .with_batch_exporter(exporter)
+            .with_resource(
+                Resource::builder_empty()
+                    .with_service_name(telemetry_config.service_name.clone())
+                    .with_attribute(KeyValue::new(
+                        "service.version",
+                        env!("CARGO_PKG_VERSION"),
+                    ))
+                    .build(),
+            )
+            .build();
+
+        // keep provider alive for the whole program
+        LOGGER_PROVIDER.set(provider.clone()).ok();
```

Comment:
If I didn't do this the provider whent out of scoped and it was dropped. This keeps it running while the program is running and allows to invoke the shutdown afterwards